ros/angel_system_nodes/angel_system_nodes/object_detector_with_descriptors.py

```diff
@@ -276,37 +276,21 @@ def postprocess_detections(self, rois, cls_prob, pooled_feat, im_scales):
         scores = cls_prob.data
         boxes = rois.data[:, :, 1:5]
 
+        boxes /= im_scales[0]
         # Simply repeat the boxes, once for each class
-        pred_boxes = np.tile(boxes.cpu(), (1, scores.cpu().shape[2]))
-        pred_boxes /= im_scales[0]
-
         scores = scores.squeeze()
-        pred_boxes = pred_boxes.squeeze()
-
-        max_conf = torch.zeros((pred_boxes.shape[0])).to(device=self._torch_device)
-
-        for j in range(1, len(self.classes)):
-            inds = torch.nonzero(scores[:,j]>self._detection_threshold).view(-1).cpu()
-            # if there is det
-            if inds.numel() > 0:
-                cls_scores = scores[:,j][inds]
-                _, order = torch.sort(cls_scores, 0, True)
-                boxs_inds = pred_boxes[inds]
-                if boxs_inds.ndim == 1:
-                    boxs_inds = np.expand_dims(boxs_inds, 0)
-                cls_boxes = (
-                    torch.tensor(boxs_inds[:, j * 4:(j + 1) * 4])
-                    .to(device=self._torch_device)
-                )
-
-                keep = nms(cls_boxes[order, :], cls_scores[order], 0.3)
-                index = inds[order[keep]]
-                max_conf[index] = (
-                    torch.where(scores[index, j] > max_conf[index],
-                                scores[index, j],
-                                max_conf[index])
-                )
-
+        max_conf = torch.zeros((scores.shape[0])).to(device=self._torch_device)
+        inds2 = torch.nonzero(scores >self._detection_threshold, as_tuple=True)
+        cls_scores = scores[inds2]
+        _, order = torch.sort(cls_scores, 0, True)
+        filled_boxs = boxes[:,inds2[0],:].to(device=self._torch_device)
```

Comment:
The previous code was casting `boxes` to the CPU. Is that tensor already on the GPU at this point? Does calling `.to(...)` cause any extraneous device copies?